R/extremes-data.R

```diff
@@ -7,47 +7,73 @@
+    minRows <- append(minRows, createDataRows(data[[which(names(data) %in% c("upchain"))]], "min", paste("Min Inst ", upchainParameter, " and corresponding ", primaryParameter), TRUE))
+    minRows <- append(minRows, createDataRows(data[[which(names(data) %in% c("primary"))]], "min", paste("Min Inst ", primaryParameter, " and corresponding ", upchainParameter), FALSE))
+  } else {
+    maxRows <- append(maxRows, createDataRows(data[[which(names(data) %in% c("primary"))]], "max", paste("Max Inst ", primaryParameter), FALSE, FALSE))
+    minRows <- append(minRows, createDataRows(data[[which(names(data) %in% c("primary"))]], "min", paste("Min Inst ", primaryParameter), FALSE, FALSE))
+  }
   
-  orderedRowNames <- c(paste("Max Inst ", upchainParameter, " and corresponding ", primaryParameter), 
-                       paste("Max Inst ", primaryParameter, " and corresponding ", upchainParameter),
-                       paste("Max Daily ", dvComputation, " ", dvParameter),
-                       paste("Min Inst ", upchainParameter, " and corresponding ", primaryParameter),
-                       paste("Min Inst ", primaryParameter, " and corresponding ", upchainParameter),
-                       paste("Min Daily ", dvComputation, " ", dvParameter)
-                       )
+  if(!no_dv){
+    dvLabel <- getReportMetadata(rawData,'dvLabel')
+    dvParameter <- getReportMetadata(rawData,'dvParameter')
+    dvComputation <- getReportMetadata(rawData,'dvComputation')
+    dvUnit <- getReportMetadata(rawData,'dvUnit')
+    if(!no_upchain){
+      maxRows <- append(maxRows, createDataRows(data[[which(names(data) %in% c("dv"))]], "max", paste("Max Daily ", dvComputation, " ", dvParameter), FALSE))
+      minRows <- append(minRows, createDataRows(data[[which(names(data) %in% c("dv"))]], "min", paste("Min Daily ", dvComputation, " ", dvParameter), FALSE))
+    } else {
+      maxRows <- append(maxRows, createDataRows(data[[which(names(data) %in% c("dv"))]], "max", paste("Max Daily ", dvComputation, " ", dvParameter), FALSE, FALSE))
+      minRows <- append(minRows, createDataRows(data[[which(names(data) %in% c("dv"))]], "min", paste("Min Daily ", dvComputation, " ", dvParameter), FALSE, FALSE))
+    }
+  }
+
+  dataRows <- c(maxRows, minRows)
+
+  #Change column and row names to their correct forms and add them into the dataframe.
+  toRet <- data.frame()
   
-  index <- which(names(data) %in% c("upchain", "primary", "dv")) 
-  results <- list()
+  for(i in 1:length(dataRows)){
+    toAdd <- cbind(dataRows[[i]][["name"]],dataRows[[i]][,-1]) 
+    colnames(toAdd) <- columnNames
+    toRet <- rbind(toRet,toAdd)
+  }
   
-  for (i in index) {  
-    
-    subset <- data[[i]][which(names(data[[i]])%in%c("min","max"))]
-    
-    min.max <- lapply(subset, function(x) {
+  return(toRet)
+}
+
+createDataRows <- function(data, param, rowName, isUpchain, includeRelated=TRUE, doMerge=TRUE){  
+    subset <- data[which(names(data)%in%c(param))]
```

Comment:
rename subset to something else (maybe `subsetData`) because subset is a function name and you are overwriting here. You can check if the name already is a function by typing it and then hitting tab in the console - if no suggestions pop up then you're good